pkg/assembler/backends/inmem/certifyVEXStatement.go

```diff
@@ -501,13 +523,30 @@ func (c *demoClient) buildCertifyVEXStatement(link *vexLink, filter *model.Certi
 	}
 
 	certifyVuln := model.CertifyVEXStatement{
-		ID:            nodeID(link.id),
-		Subject:       subj,
-		Vulnerability: vuln,
-		Justification: link.justification,
-		KnownSince:    link.knownSince,
-		Origin:        link.origin,
-		Collector:     link.collector,
+		ID:               nodeID(link.id),
+		Subject:          subj,
+		Vulnerability:    vuln,
+		Status:           link.status,
+		VexJustification: link.justification,
+		Statement:        link.statement,
+		StatusNotes:      link.statusNotes,
+		KnownSince:       link.knownSince,
+		Origin:           link.origin,
+		Collector:        link.collector,
 	}
 	return &certifyVuln, nil
 }
+
+/*
+For [status] “not_affected”, a VEX statement SHOULD provide a [justification].
+If [justification] is not provided then [impact_statement] MUST be provided.
+For [status] “affected”, MUST include one [action_statement]
+*/
+func validateVexInputBasedOnStatus(status model.VexStatus, justification model.VexJustification, statement string) error {
+	if status == model.VexStatusNotAffected && justification == model.VexJustificationNotProvided && statement == "" {
+		return fmt.Errorf("for [status] “not_affected”, if [justification] is not provided then [impact_statement] MUST be provided")
+	} else if status == model.VexStatusAffected && justification == model.VexJustificationNotProvided && statement == "" {
+		return fmt.Errorf("for [status] “affected”, MUST include one [action_statement]")
```

Comment:
minor nit here probably should have it as `statement` since the API doesn't expose that. otherwise lgtm